src/Cardano/Wallet.hs

```diff
@@ -94,14 +129,27 @@ data NewWallet = NewWallet
 
 -- | Errors occuring when fetching a wallet
 newtype ReadWalletError
-    = ErrReadWalletNotFound WalletId
+    = ReadWalletErrorNotFound WalletId
```

Comment:
The is the part where everything is a huge mess.

Multiple operations have can throw `WalletNotFound`. With one error-type per operation, we need to disambiguate between multiple `ReadWalletErrors`. Thus I made sure to prefix all constructors with the type-name.

> CreateUnsignedTxReadWalletError ReadWalletError

I also reused `ReadWalletError` where possible, because I thought we should be able to reuse formatting when presenting it in the api.

Reply:
I am wondering if we shouldn't do something completely different though… 🤔 

Maybe something like:

```haskell
class CreateWalletError
instance ErrWalletAlreadyExists CreateWalletError
```

Reply:
Running into the same problem, I've tried a different approach in #158, aligned with what we've been doing so far: https://github.com/input-output-hk/cardano-wallet/pull/158/files#diff-e2b4e9042c03ab5c8f8b121a8f1b6672L73

What do you think ?

Reply:
So the errors are the same, semantically speaking, but we still have different type per operation to get an actual type-safety and not conflate all errors altogether.

Reply:
Interesting. But I have
```haskell
-- | Errors occuring when creating an unsigned transaction
data CreateUnsignedTxError
    = CreateUnsignedTxReadWalletError ReadWalletError
    | CreateUnsignedTxCoinSelectionError CoinSelectionError

-- | Errors occuring when signing a transaction
data SignTxError
    = SignTxReadWalletError ReadWalletError
    | SignTxError
```

where the problem is the duplication of the `[...]ReadWalletError` constructor.

If the other cases —`SignTx` and `CreateUnsignedTxCoinSelectionError ` — hadn't existed your tagged `ErrNoSuchWallet "readWallet"` would have been perfect.

Reply:
I did consider using
```haskell
data ReadWalletError otherError = ErrNoSuchWallet WalletId | Other otherError
```

e.g like
```haskell
ExceptT (ReadWalletError Void)
```

but that felt a bit meh.

Reply:
Hmm.. No, and it'd defy the purpose here. It's fine to have:

```hs
data SignTxError
    = SignTxReadWalletError ReadWalletError
    | SignTxError
```

We don't necessarily need to have different `ReadWalletError` for each operation actually, since they really represent the same error here. Though, we definitely do NOT want to have one `WalletLayerError` big ADT that would conflate them all. 

What really matters here is to be _true_ about what errors a function can return.